File: Export/Natives/Name/SeparatorFileNamePartProvider.cs


using System;
using kCura.WinEDDS.Core.Model;
using kCura.WinEDDS.Exporters;

namespace kCura.WinEDDS.Core.Export.Natives.Name
{
	/// <summary>
	/// This class stores field artifact Id that will be used to create native file name during export process
	/// </summary>
	/// <typeparam name="T">field artifact id</typeparam>
	public class SeparatorFileNamePartProvider : FileNamePartProvider<SeparatorDescriptorPart>
	{
		public override string GetPartName(SeparatorDescriptorPart descriptorDescriptorPart, ObjectExportInfo exportObject)
		{
			return descriptorDescriptorPart.Value;
		}
	}
}

File: Source/Relativity.DataExchange.Legacy/FileNaming/CustomFileNaming/SeparatorFileNamePartProvider.vb
Imports FileNaming.CustomFileNaming
Imports kCura.WinEDDS.Exporters

Namespace kCura.WinEDDS.FileNaming.CustomFileNaming
	Public Class SeparatorFileNamePartProvider
		Inherits FileNamePartProvider(Of SeparatorDescriptorPart)

		Public Overrides Function GetPartName(descriptorPart As SeparatorDescriptorPart, exportObject As ObjectExportInfo) As String
			Return descriptorPart.Value
		End Function

	End Class
End Namespace
